ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineContext.java

```diff
@@ -208,16 +231,90 @@ DdlCommand createDdlCommand(
   String executeDdl(
       final String sqlExpression,
       final DdlCommand command,
-      final boolean withQuery
+      final boolean withQuery,
+      final Set<SourceName> withQuerySources
   ) {
-    final DdlCommandResult result = ddlCommandExec.execute(sqlExpression, command, withQuery);
+    if (command instanceof DropSourceCommand) {
+      throwIfOtherQueriesExist(((DropSourceCommand) command).getSourceName());
+    }
+
+    final DdlCommandResult result =
+        ddlCommandExec.execute(sqlExpression, command, withQuery, withQuerySources);
     if (!result.isSuccess()) {
       throw new KsqlStatementException(result.getMessage(), sqlExpression);
     }
+
+    if (command instanceof DropSourceCommand) {
```

Comment:
nit: can we add a comment explaining why we terminate the CAS query after we drop the source? It's not clear for me reading the code (without looking at other files) why we don't do it before.

Reply:
Done